src/matrices.jl

```diff
@@ -1,15 +1,24 @@
-@inline function pm_Matrix{T}(mat::AbstractMatrix) where T
+const pm_Matrix_suppT = Union{pm_Integer, pm_Rational, Float64}
+
+@inline function pm_Matrix{T}(mat::AbstractMatrix) where T <: pm_Matrix_suppT
     res = pm_Matrix{T}(size(mat)...)
     @inbounds res .= mat
     return res
 end
 
 # needed for broadcast
-@inline function pm_Matrix{T}(rows::UR, cols::UR) where {T, UR<:Base.AbstractUnitRange}
+@inline function pm_Matrix{T}(rows::UR, cols::UR) where {T <: pm_Matrix_suppT, UR<:Base.AbstractUnitRange}
```

Comment:
Q: Can we only participate with unit ranges?

Reply:
not sure what do you mean by participate? this is needed for similar, which in turn is needed for broadcasting

Reply:
But what about broadcasting with non unit ranges, i.e., `1:2:9`? (But maybe not that important)

Reply:
I added this a while ago when I wanted to make `f.(::pm_Matrix)` return a `pm_Matrix` instead of `Array{..., 2}`. It seems that we don't need non-unital ranges for this.